python/tank/util/shotgun.py

```diff
@@ -215,8 +216,10 @@ def __create_sg_connection(config_data=None):
             # This is needed for backwards compatibility with scripts that were
             # written before authentication was put in place. Since those scripts
             # don't set the current user, we have to get the one configured for the
-            # core instead.
-            user = sa.get_user()
+            # core instead. If the script user is configured, then the
+            # CoreDefaultsManager will provide its credentials for us and
+            # sa.get_default_user will return the ScriptUser instance.
+            user = sa.get_default_user()
```

Comment:
Calling set_current_user(user) means that now we have a current user and that serializing the context would now include the script user credentials. I don't think it would break anything, but it clearly would have a side effect.